lib/checks/navigation/region-after.js

```diff
@@ -0,0 +1,24 @@
+function regionAfter(results) {
+  const iframesInRegions = [];
+
+  results.forEach(r => {
+    if (r.data && r.data.nodeName === 'iframe') {
+      // if result is true, the rule detected the iframe was in a region
+      if (r.result) {
+        iframesInRegions.push(r.node.ancestry);
+      }
+      // pass every iframe element
+      r.result = true;
+    } else if (!r.result) {
+      // check if any node that failed is contained in an iframe that's in a region
+      // if it does, the node should pass
+      r.result = iframesInRegions.some(iframe => {
+        return iframe.every((value, index) => value === r.node.ancestry[index]);
```

Comment:
I probably should have thought of this yesterday, but this isn't going to work for iframes nested in shadow DOM trees. There the `value` can be another array, not just a string.